src/main/scala/org/zalando/react/nakadi/ReactiveNakadi.scala

```diff
@@ -86,6 +96,66 @@ class ReactiveNakadi {
     consumeWithOffsetSink(props, ReactiveNakadi.ConsumerDefaultDispatcher)
   }
 
+  def producerFlow(producerProperties: ProducerProperties)(implicit system: ActorSystem): Flow[ProducerMessage, Seq[String], NotUsed] = {
+    import org.zalando.react.nakadi.client.models.JsonOps._
+
+
+    import akka.http.scaladsl.model.MediaTypes._
+    import akka.http.scaladsl.model.headers._
+    import concurrent.duration._
+
+    implicit val materializer = ActorMaterializer()
+
+    val superPool = Http().superPool[Seq[String]]()
+
+    val uri = s"${producerProperties.serverProperties}${URI_POST_EVENTS.format(producerProperties.eventType)}"
+
+    val requestWithoutToken = HttpRequest(
+      method = HttpMethods.POST,
+      uri = Uri(uri),
+      headers = List(Accept(MediaRange(`application/json`)))
+    )
+
+    implicit val ex = system.dispatcher
+
+    val request = producerProperties.tokenProvider.fold(requestWithoutToken) { tokenFactory =>
+      requestWithoutToken.addHeader(Authorization(OAuth2BearerToken(tokenFactory())))
+    }
+
+    val flow: Flow[ProducerMessage, Seq[String], NotUsed] = Flow[ProducerMessage]
+      .map { case ProducerMessage(eventRecords) =>
+        val requestWithEntity = request.withEntity(
+          HttpEntity(ContentType(`application/json`),
+            Json.toJson(eventRecords).toString)
+        )
+
+        eventRecords.headOption.flatMap(_.metadata.flow_id).fold(requestWithEntity) { flowId =>
+          requestWithEntity.withHeaders(RawHeader("X-Flow-Id", flowId))
+        } -> eventRecords.map(_.metadata.eid)
+      }
+      .via(superPool)
+        .mapAsync(4) {
+          case (Success(response), eventIds) if response.status.isSuccess() =>
+            response.entity.dataBytes.runWith(Sink.ignore) map (_ => eventIds)
+
+          case (Success(response), eventIds) =>
+            response.entity.toStrict(1.second).flatMap { entity =>
```

Comment:
What are the implications if the timeout of 1 second is reached on logic further down the flow of the stream?